crypto/s2n_ecdsa.c

```diff
@@ -32,8 +35,29 @@
 #include "crypto/s2n_openssl.h"
 #include "crypto/s2n_pkey.h"
 
+
 #define S2N_ECDSA_TYPE 0
 
+S2N_RESULT s2n_unsafe_ecdsa_get_mut(const struct s2n_ecdsa_key *ecdsa_key, EC_KEY **out_ec_key) {
+    RESULT_ENSURE_REF(ecdsa_key);
+    RESULT_ENSURE_REF(ecdsa_key->ec_key);
+    RESULT_ENSURE_EQ(*out_ec_key, NULL);
+
+    /* pragma gcc diagnostic was added in gcc 4.6 */
+#if S2N_GCC_VERSION_AT_LEAST_OR_CLANG(4,6,0)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
+    *out_ec_key = (EC_KEY *) ecdsa_key->ec_key;
```

Comment:
why is this cast necessary?

Reply:
the point of this fn is to convert
`(const EC_KEY *) ecdsa_key->ec_key` -> `(EC_KEY *) ecdsa_key->ec_key`